### PR TITLE
MRG: update docs for v0.9.0

### PR DESCRIPTION
**Pull request by ctb**
After https://github.com/sourmash-bio/sourmash_plugin_branchwater/pull/197 and https://github.com/sourmash-bio/sourmash_plugin_branchwater/pull/211, we need to switch up our documentation and references for using the branchwater plugin.

This PR also adds documentation for `pairwise` and suggests `.rocksdb` as the suffix for `index` output.

Finally, this PR bumps the version to v0.9.0 in preparation for a release. 🎉 

The updated `doc/README.md` is [here](https://github.com/sourmash-bio/sourmash_plugin_branchwater/blob/update_docs2/doc/README.md).

HackMD version of `doc/README.md` for nicer viewing/editing: https://hackmd.io/_DqWfa8BS364lfjhnvfnOg?view

- Fixes https://github.com/sourmash-bio/sourmash_plugin_branchwater/issues/212
- Fixes https://github.com/sourmash-bio/sourmash_plugin_branchwater/issues/210
- Fixes https://github.com/sourmash-bio/sourmash_plugin_branchwater/issues/190

### CODE REVIEW

**Comment by ctb** [comment]
hmm.
```
  × Preparing editable metadata (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [15 lines of output]
      error: failed to select a version for `env_logger`.
          ... required by package `sourmash_plugin_branchwater v0.8.7-dev (/home/ctbrown/pyo3_branchwater)`
      versions that meet the requirements `^0.11.1` (locked to 0.11.1) are: 0.11.1
     
      the package `sourmash_plugin_branchwater` depends on `env_logger`, with features: `anstream` but `env_logger` does not have these features.
       It has an optional dependency with that name, but that dependency uses the "dep:" syntax in the features table, so it does not have an implicit feature with that name.
     
     
      failed to select a version for `env_logger` which could resolve this conflict
      � maturin failed
        Caused by: Cargo metadata failed. Does your crate compile with `cargo build`?
        Caused by: `cargo metadata` exited with an error:
      Error running maturin: Command '['maturin', 'pep517', 'write-dist-info', '--metadata-directory', '/scratch/tmp/pip-modern-metadata-bkog6pe2', '--interpreter', '/home/ctbrown/mambaforge/envs/sourmash-dev/bin/python']' returned non-zero exit status 1.
      Checking for Rust toolchain....
      Running `maturin pep517 write-dist-info --metadata-directory /scratch/tmp/pip-modern-metadata-bkog6pe2 --interpreter /home/ctbrown/mambaforge/envs/sourmash-dev/bin/python`
      [end of output]
```

**Comment by ctb** [comment]
maybe broken by https://github.com/sourmash-bio/sourmash_plugin_branchwater/pull/209 - will try reverting.

**Comment by ctb** [comment]
(reversion works)

**Comment by ctb** [comment]
Ready for review & merge @bluegenes!

File: src/utils.rs [hunk 1]
```diff
@@ -466,7 +466,7 @@ pub fn load_collection(
     };
 
     let collection = collection.or_else(|| {
-        dbg!("attempting to load as manifest");
+        // dbg!("attempting to load as manifest");
```
Comment (bluegenes) [inline src/utils.rs:469]:
```suggestion
```

File: src/utils.rs [hunk 2]
```diff
@@ -477,7 +477,7 @@ pub fn load_collection(
     });
 
     let collection = collection.or_else(|| {
-        dbg!("attempting to load as signature");
+        // dbg!("attempting to load as signature");
```
Comment (bluegenes) [inline src/utils.rs:480]:
```suggestion
```

File: src/utils.rs [hunk 3]
```diff
@@ -488,7 +488,7 @@ pub fn load_collection(
     });
 
     let collection = collection.or_else(|| {
-        dbg!("attempting to load as pathlist");
+        // dbg!("attempting to load as pathlist");
```
Comment (bluegenes) [inline src/utils.rs:491]:
```suggestion
```

File: Cargo.toml [hunk 4]
```diff
@@ -16,7 +16,7 @@ sourmash = { version = "0.12.1", features = ["branchwater"] }
 serde_json = "1.0.113"
 niffler = "2.4.0"
 log = "0.4.14"
-env_logger = "0.11.1"
+env_logger = "0.10.2"
```
Comment (bluegenes) [inline Cargo.toml:19]: backing up on purpose?

**Review by bluegenes — APPROVED** [review]
made a few suggestions -- otherwise looks good!!

File: Cargo.toml (hunk 4 above)
Comment (ctb) [reply Cargo.toml:19]: yes, it causes [build problems on linux](https://github.com/sourmash-bio/sourmash_plugin_branchwater/pull/213#issuecomment-1946239336)

**Pull request merged**